sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/InsertAdaptiveSparkPlan.scala

```diff
@@ -120,7 +124,7 @@ case class InsertAdaptiveSparkPlan(session: SparkSession) extends Rule[SparkPlan
     val queryExec = new QueryExecution(session, plan)
     // Apply the same instance of this rule to sub-queries so that sub-queries all share the
     // same `stageCache` for Exchange reuse.
-    val adaptivePlan = this.apply(queryExec.sparkPlan)
+    val adaptivePlan = this.applyInternal(queryExec.sparkPlan, queryExec)
```

Comment:
When we reach here, it means we are creating `AdaptiveSparkPlanExec` for subqueries. Shall we simply set a boolean flag here (e.g. `adaptivePlan.copy(isSubquery = true)`) instead of passing around the `QueryExecution`?

Reply:
nvm, this is more flexible, in case some places create `QueryExecution` without execution id and execute.